src/main/java/frc/robot/subsystems/ball/feeder/UltrasonicPoller.java

```diff
@@ -9,22 +10,50 @@
 class UltrasonicPoller extends RepeatingPooledSubsystem {
 
     Ultrasonic ultrasonicSensor;
+    StopWatch timer = new StopWatch();
 
-    UltrasonicPoller(Ultrasonic sensor, int time) {
+    // Consider renaming, couldn't think of a better one ATM
+    boolean noBalls;
+    long time;
+    double distance;
+    double currentRange;
+
+    UltrasonicPoller(Ultrasonic sensor, long time, double distance, long rate) {
         // sensor is pretty self explanatory
         // time is how long you want the sensor to not detect an object until true is returned
-        super(25, TimeUnit.MILLISECONDS); // 4Hz seems reasonable
+        // distance is the minimum distance threshold
+        super(rate, TimeUnit.MILLISECONDS); // 4Hz seems reasonable
         this.ultrasonicSensor = sensor;
+        this.time = time;
+        this.distance = distance;
+        this.noBalls = false;
+
+    }
+
+    UltrasonicPoller(Ultrasonic sensor, long time, double distance) {
+        this(sensor, time, distance, 25);
     }
 
     @Override
     public void defineResources(){
         require(ultrasonicSensor);
     }
 
+    // Consider renaming, probably want to generalize the name if we ever re-use it
+    public boolean checkUnloaded(){
+        return this.noBalls;
+    }
+
     @Override
     public void task(){
-        // Placholder
-        return;
+        if(!this.timer.isStarted()){
+            this.timer.start();
+        }
+        this.currentRange = this.ultrasonicSensor.getRangeMM();
+        if((this.currentRange > this.distance) && (this.timer.getTime() > this.time)) {
+            this.noBalls = true;
```

Comment:
`noBalls` is initially `false` and becomes `true` under this condition. When can it transition back to being `false`?